src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/BundleSlingInitialContentJarEntryExtractor.java

```diff
@@ -84,7 +82,7 @@ void extractAndAddToAssembler(@NotNull BundleSlingInitialContentExtractContext c
             VaultPackageAssembler packageAssembler = assemblerProvider.initPackageAssemblerForPath(context, repositoryPath, pathEntryValue);
 
             final ContentReader contentReader = contentReaderProvider.getContentReaderForEntry(file, pathEntryValue);
-            if (contentReader != null) {
+            if (contentReader != null && !isDefaultContentXmlFile) {
```

Comment:
if `isDefaultContentXmlFile` is false you don't need to obtain the `contentReader` to start with, no?
i am not sure i understand why this is the case, but it would make the code more readable if you would re-arrange it and only retrieve the contentreader if needed.